Move metrics to slog

aanm opened the pull request:
Because of these two log lines (59 and 73) https://github.com/cilium/cilium/blob/3d844c22ad5c06d4e56e262ddbd0d40f49989b23/pkg/metrics/metric/metric.go#L58-L77 we had to change all of the call paths to pass down the respective loggers into the "newMetric" functions.

I'll be happy to revert back these changes if we don't need to warn the users. Otherwise we keep the code as it is and the `*slog.Logger` is a mandatory argument of the NewMetrics function.

tommyp1ckles commented on pkg/k8s/resource/resource.go:
```diff
@@ -822,3 +825,33 @@ func getUID(obj k8sRuntime.Object) types.UID {
+type WorkqueueMetricsProvider struct{}
+
+func (WorkqueueMetricsProvider) NewDepthMetric(name string) workqueue.GaugeMetric {
+	return metrics.WorkQueueDepth.WithLabelValues(name)
```
Won't this global just always be Noop?  Trying this out locally this seems to break the metrics as these are not part of the registry being used to gather metrics.

aanm committed:
pkg/metrics: migrate to slog

Migrate this package to use slog instead of logrus.

Signed-off-by: André Martins <andre@cilium.io>

aanm committed:
pkg/metrics: enable invalid label values detector in CI

Since we are going to panic in the CI every time a label has an unknown
value we should fix existing labels that have unknown values.

Signed-off-by: André Martins <andre@cilium.io>

tommyp1ckles reviewed (approved):
nice thanks!
